hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2209,14 +2205,18 @@ private void listStatusFindKeyInTableCache(
           bucketName);
     }
 
+    List<OmKeyInfo> keyInfoList = new ArrayList<>(fileStatusList.size());
     for (OzoneFileStatus fileStatus : fileStatusList) {
-      if (args.getRefreshPipeline()) {
-        refreshPipeline(fileStatus.getKeyInfo());
-      }
-      if (args.getSortDatanodes()) {
+      keyInfoList.add(fileStatus.getKeyInfo());
+    }
+    refreshPipeline(keyInfoList);
```

Comment:
Just a question listStatus is limited by batch size.
But we are making a single Rpc Call to SCM for all the containers we got from this list, is this fine here? (My question is if it has a large number of containerIDs due to larger key sizes, will it have an issue in Rpc Response size.

Reply:
`listKeys()` is also limited by batch size and already does the same single refresh call:

https://github.com/apache/ozone/blob/d0aa34c4afae21538c6c6225f029c1d1c4c4bafd/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L931-L933

Reply:
My question is keyList is limited by batch size, but if Keys are huge size, then it might have huge containerIDs, and then calling SCM with all those containerIDS to fetch ContainerWithPipeline, will it cause an issue.

Then this might be a problem for listKeys also, but do you see this as a problem?

Reply:
> Then this might be a problem for listKeys also, but do you see this as a problem?

I don't see this as an immediate problem.

As far as I understand, batch size for `listKeys` and `listStatus` is a convenience for the client, not a safety guarantuee for the server.  If RPC response size is a problem when performing `getContainerWithPipelineBatch` call for multiple keys, then the same can be triggered by the client simply by increasing batch size (if there are enough keys in the bucket).

Reply:
Ya agreed, not only that thinking more if with normall batch size it is causing issue for containerID's, then it will be same for OMResponse for listKeys also, as it adds KeyInfo and Pipeline.